Daemon support for --follow-imports=normal

Currently the daemon only supports errors and ignore, which is a big limitation.
Additionally, while we reject follow-imports=normal as a top-level option, we
silently allow it for module-level definitions while not correctly implementing it.
(Including for all stubs, as in #5566.)

This limitation is blocking usage of the daemon with PEP 561 packages that use inline type information. However, PEP 561 packages that have type information in pyi files do work (#5566). 

This inconsistency is quite surprising. I can imagine a package moving their type info from pyi to py and suddenly ceasing to work with the daemon.

Personally, I think package imports should be followed even for the daemon. The original concern of tracking updates isn't really relevant here – installed packages rarely change (unlike user code). 

So maybe there should be a `follow_imports_for_installed_packages` option? Similar to `follow_imports_for_stubs`, it will override follow_imports to always use 'normal' for installed packages. I would argue that this should be set to False by default, but then there's backwards compat concerns (?). In any case, it seems like full `--follow-imports=normal` for the daemon will not be supported anytime soon because it's difficult to implement and isn't a common use case.



Repro steps:
```shell
$ echo "import hello" > test.py

# Package with inline type information: works with mypy, not with dmypy
$ pip install "git+https://github.com/ymyzk/pep-561-samples#subdirectory=hello-typed1"
$ mypy test.py
Success: no issues found in 1 source file
$ dmypy run -- --follow-imports=error test.py; dmypy stop
test.py:1: error: Cannot find module named 'hello'

# Package with type information in stubs: works with both mypy and dmypy
$ pip uninstall hello-typed1
$ pip install "git+https://github.com/ymyzk/pep-561-samples#subdirectory=hello-typed2"
$ mypy test.py
Success: no issues found in 1 source file
$ dmypy run -- --follow-imports=error test.py; dmypy stop
Success: no issues found in 1 source file

```

If it's useful to anyone, there is a workaround that uses the 'bug' described by @msullivan in the first post – creating `mypy.ini` with the following contents will allow the daemon to follow imports to package that use inline type info:
```ini
[mypy]
[mypy-hello]
follow_imports=normal
```

--- a/mypy/dmypy_server.py
+++ b/mypy/dmypy_server.py
@@ -143,7 +143,7 @@ def process_start_options(flags: List[str], allow_sources: bool) -> Options:
         sys.exit("dmypy: start/restart should not disable incremental mode")
     # Our file change tracking can't yet handle changes to files that aren't
     # specified in the sources list.
-    if options.follow_imports not in ('skip', 'error'):
+    if options.follow_imports not in ('skip', 'error', 'normal'):
         sys.exit("dmypy: follow-imports must be 'skip' or 'error'")
     return options
 

